engine/api/authentication/oidc/openid.go

```diff
@@ -0,0 +1,139 @@
+	provider, err := oidc.NewProvider(context.Background(), url)
+	if err != nil {
+		return nil, sdk.WrapError(err, "Failed to initialize OIDC driver")
+	}
+	// Configure an OpenID Connect aware OAuth2 client.
+	oauth2Config := oauth2.Config{
+		ClientID:     clientID,
+		ClientSecret: clientSecret,
+		RedirectURL:  fmt.Sprintf("%s/auth/callback/%s", cdsURL, sdk.ConsumerOIDC),
+		// Discovery returns the OAuth2 endpoints.
+		Endpoint: provider.Endpoint(),
+		// "openid" is a required scope for OpenID Connect flows.
+		Scopes: []string{oidc.ScopeOpenID, "profile", "email"},
+	}
+	oidcConfig := &oidc.Config{
+		ClientID: clientID,
+	}
+	verifier := provider.Verifier(oidcConfig)
+
+	return &authDriver{
+		signupDisabled: signupDisabled,
+		cdsURL:         cdsURL,
+		OAuth2Config:   oauth2Config,
+		Verifier:       verifier,
+	}, nil
+}
+
+type authDriver struct {
+	signupDisabled bool
+	cdsURL         string
+	OAuth2Config   oauth2.Config
+	Verifier       *oidc.IDTokenVerifier
+}
+
+func (d authDriver) GetManifest() sdk.AuthDriverManifest {
+	return sdk.AuthDriverManifest{
+		Type:           sdk.ConsumerOIDC,
+		SignupDisabled: d.signupDisabled,
+	}
+}
+
+func (d authDriver) GetSigninURI(signinState sdk.AuthSigninConsumerToken) (sdk.AuthDriverSigningRedirect, error) {
+	// Generate a new state value for the auth signin request
+	jws, err := authentication.NewDefaultSigninStateToken(signinState.Origin,
+		signinState.RedirectURI, signinState.IsFirstConnection)
+	if err != nil {
+		return sdk.AuthDriverSigningRedirect{}, err
+	}
+
+	var result = sdk.AuthDriverSigningRedirect{
+		Method: http.MethodGet,
+		URL:    d.OAuth2Config.AuthCodeURL(jws),
+	}
+
+	return result, nil
+}
+
+func (d authDriver) GetSessionDuration() time.Duration {
+	return time.Hour * 24 * 30 // 1 month session
+}
+
+func (d authDriver) CheckSigninRequest(req sdk.AuthConsumerSigninRequest) error {
+	if code, ok := req["code"]; !ok || code == "" {
+		return sdk.NewErrorFrom(sdk.ErrWrongRequest, "missing or invalid code")
+	}
+	return nil
+}
+
+func (d authDriver) CheckSigninStateToken(req sdk.AuthConsumerSigninRequest) error {
+	// Check if state is given and if its valid
+	state, okState := req["state"]
+	if !okState {
+		return sdk.NewErrorFrom(sdk.ErrWrongRequest, "missing state value")
+	}
+
+	return authentication.CheckDefaultSigninStateToken(state)
+}
+
+func (d authDriver) GetUserInfo(ctx context.Context, req sdk.AuthConsumerSigninRequest) (sdk.AuthDriverUserInfo, error) {
+	var info sdk.AuthDriverUserInfo
+
+	ctx2 := context.WithValue(context.Background(), oauth2.HTTPClient, http.DefaultClient)
+	oauth2Token, err := d.OAuth2Config.Exchange(ctx2, req["code"])
+	if err != nil {
+		return info, sdk.WrapError(err, "Failed to exchange token")
+	}
+	rawIDToken, ok := oauth2Token.Extra("id_token").(string)
+	if !ok {
+		return info, sdk.WrapError(err, "No id_token field in oauth2 token")
```

Comment:
you cant' use `err`. You have to do `sdk.WithStack(fmt.Errorf("no id_token field in oauth2 token"))`.

Reply:
Copy paste mistake, thanks for spotting that